src/Helper/GridHelperService.php

```diff
@@ -591,6 +591,18 @@ public function prepareListingForGrid(array $requestParams, string $requestedLan
             $conditionFilters[] = '(`path` = ' . $quotedPath . ' OR `path` like ' . $quotedWildcardPath . ')';
         }
 
+        if (isset($requestParams['all_objects']) && $requestParams['all_objects'] === 'true') {
+            $conditionFilters[] = "(`type` = 'object' OR `type` = 'variant')";
+        } else {
```

Comment:
I added this few lines because the first time you load a folder you can correctly change the type of objects to see. But on the second change it stays stuck on the first one.

Reply:
Reply on the first comment.